pkg/test/framework/components/echo/config.go

```diff
@@ -72,6 +72,16 @@ type Config struct {
 	// ReadinessTimeout specifies the timeout that we wait the application to
 	// become ready.
 	ReadinessTimeout time.Duration
+
+	// Workloads contains the list of workloads config belonging to this echo service instance.
+	Workloads []WorkloadConfig
+}
+
+// WorkloadConfig is the config for a group of workloads (e.g. Kubernetes deployment).
+type WorkloadConfig struct {
```

Comment:
I feel that some other deployment-specific fields should be pulled out of the the top-level config and land here. WDYT?

Reply:
Agree. That's why I marked a TODO below.

Reply:
done. see testdata folder